internal/auth/auth.go

```diff
@@ -13,29 +14,31 @@ import (
 
 //go:generate mockgen -source=auth.go -package=auth -destination=mock_auth.go
 
-type RegistryAuth interface {
-	GetKeyChainFromSecret(ctx context.Context, secretName, secretNamespace string) (authn.Keychain, error)
+type RegistryAuthGetter interface {
+	GetKeyChain(ctx context.Context) (authn.Keychain, error)
 }
 
-type registryAuth struct {
-	client client.Client
+type registrySecretAuthGetter struct {
+	client         client.Client
+	namespacedName *types.NamespacedName
```

Comment:
I don't think we need to store a pointer here, how about `namespacedName` being a `types.NamespacedName`?

Reply:
> I don't think we need to store a pointer here

Why not? it is making it easier for the `if` blocks that only need to check `if registryAuthGetter.namespacedName != nil`

Reply:
Fixed.